external-crates/move/move-compiler/src/sui_mode/typing.rs

```diff
@@ -252,6 +256,19 @@ fn is_mut_clock(param_ty: &Type) -> bool {
     }
 }
 
+fn is_entry_receiving_ty(param_ty: &Type) -> bool {
+    match &param_ty.value {
+        Type_::Ref(_, t) => is_entry_receiving_ty(t),
+        Type_::Apply(_, sp!(_, n), targs)
+            if n.is(SUI_ADDR_NAME, TRANSFER_MODULE_NAME, RECEIVING_TYPE_NAME) =>
+        {
+            debug_assert!(targs.len() == 1);
+            is_entry_object_ty(&targs[0])
```

Comment:
Do we not have `T: key` on `Receiving`?

Reply:
We do not :) See my reply above to a different question which explains why it can't